### PR TITLE
Correct Kiva accelerated date calculation to account for run period start date of May 1st

### LINKED ISSUE #8835: Kiva calculations crash/segfault when the run period starts from 1st May
Issue overview
--------------
If the run period is set to start from 1st May, EP will crash (from EP launch) or give a segfault (from cli) when including Kiva. A file for reproduction is [in.idf.txt](https://github.com/NREL/EnergyPlus/files/6689601/in.idf.txt), which is identical to the example file ZoneCoupledKivaSlab.idf, except for `Begin Month` in `RunPeriod` set to 5. This issue seems to only happen with 1st May being the begin date, I didn't test all dates exhaustively tho (1st of each month and 27th Apr - 4th May were tested).

I did a quick debugging, and noticed this issue might be related to the following bits of codes, where `index` ends up being 4294967295 during the initialisation, which I suppose exceeds the index range of `kivaWeather.dryBulb`.

```cpp
   159 	    // Determine accelerated intervals
   160 	    int numAccelaratedTimesteps = 3;
   161 	    int acceleratedTimestep = 30; // days
-> 162 	    int accDate =
   163 	        state.dataEnvrn->DayOfYear - 1 - acceleratedTimestep * (numAccelaratedTimesteps + 1); // date time = last timestep from the day before
   164 	    while (accDate < 0) {
   165 	        accDate = accDate + 365 + state.dataWeatherManager->LeapYearAdd;
```

```cpp
   214 	
   215 	    std::shared_ptr<Kiva::BoundaryConditions> bcs = instance.bcs;
   216 	
-> 217 	    bcs->outdoorTemp = kivaWeather.dryBulb[index] * weightNow + kivaWeather.dryBulb[indexPrev] * (1.0 - weightNow) + DataGlobalConstants::KelvinConv;
   218 	
   219 	    bcs->localWindSpeed = (kivaWeather.windSpeed[index] * weightNow + kivaWeather.windSpeed[indexPrev] * (1.0 - weightNow)) *
   220 	                          state.dataEnvrn->WeatherFileWindModCoeff *
```

### Details
Some additional details for this issue (if relevant):
 - Mac/windows
 - EP 9.5
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by matthew-larson**
Pull request overview
---------------------
 - Fixes #8835 
 - This pull request creates an inline function for the accelerated date calculation with the fix incorporated within the new function.  On May 1st, or `state.dataEnvrn->DayOfYear` = 121, `accDate` was resulting in 0 and causing EnergyPlus to crash.  The `while` statement was modified to include values of 0 to adjust the `accDate` to an acceptable value.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts

### CODE REVIEW

**Comment by Myoldmopar** [comment]
This is reasonable and unrelated to any other E+ stuff, I will take a look and build locally, assuming the conflict is easy to resolve.

File: src/EnergyPlus/HeatBalanceKivaManager.cc [hunk 1]
```diff
@@ -189,6 +183,17 @@ void KivaInstanceMap::initGround(EnergyPlusData &state, const KivaWeatherData &k
     instance.foundation->numericalScheme = Kiva::Foundation::NS_ADI;
 }
 
+int KivaInstanceMap::getAccDate(EnergyPlusData &state, const int numAccelaratedTimesteps, const int acceleratedTimestep)
+{
+    // Determine accelerated intervals
+    int accDate =
+        state.dataEnvrn->DayOfYear - 1 - acceleratedTimestep * (numAccelaratedTimesteps + 1); // date time = last timestep from the day before
+    while (accDate <= 0) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/HeatBalanceKivaManager.cc:191]: Actual fix here, OK.

File: tst/EnergyPlus/unit/HeatBalanceKivaManager.unit.cc [hunk 2]
```diff
@@ -276,9 +277,59 @@ TEST_F(EnergyPlusFixture, OpaqueSkyCover_InterpretWeatherMissingOpaqueSkyCover)
...
+    fnd.polygon.outer().push_back(Kiva::Point(-6.0, -6.0));
+    fnd.polygon.outer().push_back(Kiva::Point(-6.0, 6.0));
+    fnd.polygon.outer().push_back(Kiva::Point(6.0, 6.0));
+    fnd.polygon.outer().push_back(Kiva::Point(6.0, -6.0));
+
+    HeatBalanceKivaManager::KivaManager km;
+    HeatBalanceKivaManager::KivaInstanceMap kv(*state, fnd, 0, {}, 0, 15.0, 1.0, 0, &km);
+
+    // Set day of year to 121 (May 1st)
+    state->dataEnvrn->DayOfYear = 121;
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/HeatBalanceKivaManager.unit.cc:348]: Simulates the defect here.

File: tst/EnergyPlus/unit/HeatBalanceKivaManager.unit.cc [hunk 3]
```diff
@@ -276,9 +277,59 @@ TEST_F(EnergyPlusFixture, OpaqueSkyCover_InterpretWeatherMissingOpaqueSkyCover)
...
+
+    int numAccelaratedTimesteps = 3;
+    int acceleratedTimestep = 30; // days
+    int accDate = kv.getAccDate(*state, numAccelaratedTimesteps, acceleratedTimestep);
+
+    // Accelerated date should be greater than 0
+    EXPECT_GT(accDate, 0);
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/HeatBalanceKivaManager.unit.cc:355]: Verifies here.

**Review by Myoldmopar — APPROVED** [review]
Yeah, this looks good to go.  Building and testing locally.

**Comment by Myoldmopar** [comment]
Well, I tested it all locally fine, but I don't have write access to update this branch with conflicts resolved.  If you can click the button to let me write to the branch, I can push my conflict resolution, otherwise I'll need to you do it @matthew-larson

**Comment by matthew-larson** [comment]
Thanks @Myoldmopar, I went ahead and just resolved the conflicts myself.

File: tst/EnergyPlus/unit/HeatBalanceKivaManager.unit.cc [hunk 4]
```diff
@@ -281,12 +281,35 @@ TEST_F(EnergyPlusFixture, OpaqueSkyCover_InterpretWeatherMissingOpaqueSkyCover)
...
+    Real64 expectedValue = totalDepthOfWallBelowGrade + 1.0;
+    EXPECT_NE(initDeepGroundDepth, fnd.deepGroundDepth);
+    EXPECT_EQ(expectedValue, fnd.deepGroundDepth);
+}
+
+TEST_F(EnergyPlusFixture, HeatBalanceKiva_GetAccDate)
+{
+    // Create Kiva foundation and set parameters
+    Kiva::Foundation fnd;
+
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/HeatBalanceKivaManager.unit.cc:312]: :+1:

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Thanks @matthew-larson I pulled and ran it successfully.  Merging.

**Pull request merged**